### PR TITLE
Merge back changes

### PR DESCRIPTION
**Pull request by Raycoms**
Merge stuff from 1.11 to 1.10

Review please

### CODE REVIEW

**Comment by marvin-bitterlich** [comment]
:shipit: get it done _:D

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/581/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/main/java/com/minecolonies/coremod/util/InventoryUtils.java [hunk 1]
```diff
@@ -415,7 +415,12 @@ public static boolean takeStackInSlot(
     @Nullable
     public static ItemStack setStack(@NotNull final IInventory inventory, @Nullable final ItemStack stack)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/util/InventoryUtils.java:416]: MAJOR The Cyclomatic Complexity of this method "setStack" is 13 which is greater than 10 authorized. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: src/main/java/com/minecolonies/coremod/util/InventoryUtils.java [hunk 2]
```diff
@@ -453,6 +458,34 @@ public static ItemStack setStack(@NotNull final IInventory inventory, @Nullable
...
+    public static ItemStack setOverSizedStack(@NotNull final IInventory inventory, @Nullable final ItemStack stack)
+    {
+        int stackSize = stack.stackSize;
+        while (stackSize > 0)
+        {
+            final int itemCount = Math.min(stackSize, stack.getMaxStackSize());
+            final ItemStack items = new ItemStack(stack.getItem(), itemCount, stack.getItemDamage());
+            stackSize-=itemCount;
+            final ItemStack remainingItems = setStack(inventory,items);
+            stackSize+=remainingItems.stackSize;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/util/InventoryUtils.java:478]: BLOCKER NullPointerException might be thrown as 'remainingItems' is nullable here [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2259)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Pull request merged**